src/plugin/helpers/validation.ts

```diff
@@ -162,7 +162,24 @@ export const required = (
   customMessage: string | undefined = undefined,
   level: ErrorLevel = "ERROR"
 ): FieldValidator => (value, field) => {
-  if (typeof value !== "string" && value !== undefined) {
+  const requiredError = {
```

Comment:
I would use ternaries and some slight rephrasing to simplify it further:

```suggestion
  const requiredError = {
    error: "Required",
    message: customMessage ?? `${field} is required`,
    level,
  };
  const typeError = {
    error: "Invalid field",
    message: `Checking if ${field} is required: value is invalid`,
    level,
  };

  const isArray = Array.isArray(value);
  const isString = typeof value === "string";

  if (isArray || isString) {
    return value.length === 0 ? [requiredError] : [];
  }

  if (value === undefined) {
    return [requiredError];
  }

  return [typeError];
```

Reply:
(the suggestion might not be covering the correct lines so would recommend against using the `Commit suggestion` button if you wanted to use this code)